cmake/Platform/Android/Toolchain_android.cmake

```diff
@@ -31,7 +31,9 @@ endif()
 if(NOT ANDROID_ABI)
     set(ANDROID_ABI arm64-v8a)
 endif()
-if(NOT ANDROID_ARM_MODE)
+
+# The ANDROID_ARM_MODE is only applicable when the ANDROID_ABI starts with "armeabi"
+if(NOT ANDROID_ARM_MODE AND ANDROID_ABI MATCHES "^armeabi")
     set(ANDROID_ARM_MODE arm)
 endif()
 if(NOT ANDROID_ARM_NEON)
```

Comment:
Alternatively, we could enforce only using **arm64-v8a** as the `ANDROID_ABI`, fatal erroring otherwise.  Realistically it's gonna fall over elsewhere anyway.  Then we could remove the setting of both `ANDROID_ARM_MODE` and `ANDROID_ARM_NEON` as they are only necessary for ARMv7 targets, like you mention.

Reply:
Doing that would also simplify the [build_config.json](https://github.com/o3de/o3de/blob/development/scripts/build/Platform/Android/build_config.json#L63) file for Android, as they also supply the `ANDROID_ARM_MODE` and `ANDROID_ARM_NEON` option